Objects/bytearrayobject.c

```diff
@@ -2900,27 +2913,39 @@ bytearrayiter_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
      * call must be before access of iterator pointers.
      * see issue #101765 */
     bytesiterobject *it = _bytesiterobject_CAST(self);
-    if (it->it_seq != NULL) {
-        return Py_BuildValue("N(O)n", iter, it->it_seq, it->it_index);
-    } else {
-        return Py_BuildValue("N(())", iter);
+    PyObject *ret = NULL;
+    Py_ssize_t index = FT_ATOMIC_LOAD_SSIZE_RELAXED(it->it_index);
+    if (index >= 0) {
+        if (index <= PyByteArray_GET_SIZE(it->it_seq)) {
```

Comment:
Is the check for `index <= PyByteArray_GET_SIZE(it->it_seq)` needed here? it wasn't there previously.

IIUC we only care if the iterator is exhausted or not which can be checked with `index >=0`.

Reply:
Yes it can be removed, I had a critical section there before to make sure `index` wasn't stored larger than the size of the sequence in the state, but it doesn't really hurt if it is so will remove.